bitcoin/bitcoin
# Unique coinbase: Fixes #482

Fix from luke-jr, I reviewed the code but have not tested.

Rough sketch of a test plan:
- Run clean testnet-in-a-box bitcoind, with -keypool=1
- Encrypt the wallet
- Run bitcoind getnewaddress until it tell you keypool is exhausted
- Generate a couple of blocks via internal miner
  -- verify:  coinbase transactions have unique txids even though they pay-to default key
- Generate a couple of blocks via getwork RPC call
  -- verify: coinbase transactions have unique txids
